src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -686,6 +776,68 @@ def cmac_algorithm_supported(self, algorithm):
     def create_cmac_ctx(self, algorithm):
         return _CMACContext(self, algorithm)
 
+    def create_x509_csr(self, builder, private_key, algorithm):
+        # TODO: check type of private key parameter.
```

Comment:
No, tests to make sure all types of keys work should just be in `test_x509` -- these are tests of the common interface, not any one implementation